### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by MaksymMaliuk**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,89 @@
...
+  shape = 'triangle';
 
-}
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
```
Comment (olecsiy-gavrilyuk) [inline src/figuresClasses.ts:16]: useless spaces

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,89 @@
...
+    return +(area.toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = 'circle';
+
+  color: string;
+
```
Comment (olecsiy-gavrilyuk) [inline src/figuresClasses.ts:44]: useless space)

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,89 @@
...
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape = 'rectangle';
+
+  color: string;
+
+  width: number;
+
+  height: number;
```
Comment (olecsiy-gavrilyuk) [inline src/figuresClasses.ts:69]: spaces))

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,89 @@
...
+
+  c: number;
 
-export class Circle {
+  constructor(color: string, a: number, b: number, c: number) {
+    if (Math.min(a, b, c) <= 0) {
+      throw new Error('Triangle sides must be greater than 0');
+    }
 
+    if (a + b <= c || a + c <= b || b + c <= a) {
```
Comment (olecsiy-gavrilyuk) [inline src/figuresClasses.ts:23]: I believe that you can do this  line more optimaze (you can create arr with sides and sort this arr after that you should only check one side)

**Review by olecsiy-gavrilyuk — APPROVED** [review]

File: src/figuresClasses.ts (hunk 1 above)
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:16]:
```suggestion
```
Use [Typescript parameter props](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties)

File: src/figuresClasses.ts [hunk 5]
```diff
@@ -1,19 +1,89 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:8]:
```suggestion
  readonly shape = 'triangle';
```
Prop shape is readonly for each figure.

Comment (sergii-nosachenko) [inline src/figuresClasses.ts:8]: Create enum for shapes

File: src/figuresClasses.ts [hunk 6]
```diff
@@ -1,19 +1,89 @@
...
+      throw new Error('Triangle sides are invalid');
+    }
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:33]: What is `p`?

File: src/figuresClasses.ts [hunk 7]
```diff
@@ -1,19 +1,89 @@
...
+      throw new Error('Circle radius must be greater than 0');
+    }
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:58]: Why do you use a different way of rounding for each figure? Create a helper function that will return a correctly rounded result for any figure.

File: src/figuresClasses.ts [hunk 8]
```diff
@@ -1,19 +1,89 @@
...
+  shape = 'rectangle';
+
+  color: string;
+
+  width: number;
+
+  height: number;
+
+  constructor(color: string, width: number, height: number) {
+    if (width <= 0 || height <= 0) {
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:72]: Create a helper function that accepts any number of args and checks if all values are correct. Reuse with all figures.

File: src/figuresClasses.ts [hunk 9]
```diff
@@ -1,19 +1,89 @@
...
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
-export class Circle {
+  constructor(color: string, a: number, b: number, c: number) {
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:18]: In case of more than 3 args place each one on the new line.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/figuresClasses.ts (hunk 9 above)
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:18]: Color is not any string. Read the task requirements carefully.

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**